spec/lib/vanagon/engine/pooler_spec.rb

```diff
@@ -21,28 +21,86 @@
   before :each do
     # suppress `#warn` output during tests
     allow_any_instance_of(Vanagon::Platform::DSL).to receive(:warn)
+
+    # stubbing ENV doesn't actually intercept ENV because ENV
```

Comment:
hmm..So we do this in one of our puppet modules spec tests...would it work for this use case?

```
allow(ENV).to receive(:[]).and_call_original
allow(ENV).to receive(:[]).with("PE_NODE_GROUP_CLASSIFICATION_ATTEMPTS").and_return("2")
```

https://github.com/puppetlabs/puppetlabs-pe_install/blob/ea403f29235fcd568a2f5f66d8422956d8c45f39/spec/unit/util/pe_node_groups.rb#L137-L138